packages/visualizations/src/types.ts

```diff
@@ -1,15 +1,28 @@
 import { ColumnCursor, IteratableFrame } from "@operational/frame";
 import { ScaleBand, ScaleLinear } from "d3-scale";
 
-export interface AxialChartProps<Name extends string> {
+export interface BaseAxialChartProps<Name extends string> {
   metricDirection: "horizontal" | "vertical";
   data: IteratableFrame<Name>;
   metric: ColumnCursor<Name>;
   categorical: ColumnCursor<Name>;
   metricScale: ScaleLinear<any, any>;
   categoricalScale: ScaleBand<string>;
   transform?: React.SVGAttributes<SVGRectElement>["transform"];
-  style?: React.SVGAttributes<SVGGElement>["style"] | ((i: number) => React.SVGAttributes<SVGGElement>["style"]);
 }
 
+export type DiscreteAxialChartProps<Name extends string> = BaseAxialChartProps<Name> & {
+  style?: React.SVGAttributes<SVGGElement>["style"] | ((d: any[], i?: number) => React.SVGAttributes<SVGGElement>["style"]);
```

Comment:
```suggestion
  style?: React.SVGAttributes<SVGGElement>["style"] | ((d: RawRow, i?: number) => React.SVGAttributes<SVGGElement>["style"]);
```

Did I understood it right? `d` meant to be row or not?

Reply:
Ah, yes, you're right. Slipped back into d3 notation there... 😉